src/sql/src/pure.rs

```diff
@@ -47,6 +48,7 @@ use crate::normalize;
 /// [`Catalog`](crate::catalog::Catalog), as that would require locking access
 /// to the catalog for an unbounded amount of time.
 pub fn purify(
+    scx: &StatementContext,
```

Comment:
So passing a `StatementContext` to purify is not the intended use of `StatementContext`. (Weird to pass in param types, and it duplicates the catalog). I think just check `catalog.config().experimental_mode` in the if?

Reply:
I changed it to use the catalog directly. For my education, what is the intended use of the various context objects?